crates/biome_js_analyze/src/analyzers/nursery/no_restricted_imports.rs

```diff
@@ -0,0 +1,77 @@
+use biome_analyze::context::RuleContext;
+use biome_analyze::{declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_deserialize_macros::Deserializable;
+use biome_js_syntax::inner_string_text;
+use biome_rowan::TextRange;
+use serde::{Deserialize, Serialize};
+
+use super::use_nodejs_import_protocol::AnyJsImportLike;
+
+declare_rule! {
+    /// Disallow specified modules when loaded by import or require.
+    ///
+    /// https://eslint.org/docs/latest/rules/no-restricted-imports
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```json
+    /// {
+    ///     "noRestrictedImports": {
+    ///         "options": {
+    ///             "paths": ["lodash"]
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ```js
+    /// const lodash = require("lodash");
+    /// ```
+    pub(crate) NoRestrictedImports {
+        version: "next",
+        name: "noRestrictedImports",
+        source: RuleSource::Eslint("no-restricted-imports"),
+        recommended: false,
+    }
+}
+
+/// Options for the rule `noRestrictedImports`.
+#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct RestrictedImportsOptions {
+    /// A list of names that should trigger the rule
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    paths: Vec<String>,
```

Comment:
We ould use a set instead:

```suggestion
    paths: FxHashSet<String>,
```

Reply:
Alternatively we could use a map for adding a custom message:

```suggestion
    paths: FxHashMap<String, String>,
```

For example:

```json
{
  "noRestrictedImports": {
    "options": {
      "paths": {
        "mod1": "use mod2 instead"
      }
    }
  }
}
```